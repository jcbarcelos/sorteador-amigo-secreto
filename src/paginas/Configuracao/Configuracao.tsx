import React from 'react'
import Card from '../../components/Card'
import Form from '../../components/Form/Form'
import { ListaParticipantes } from '../../components/ListaParticipantes/ListaParticipantes'
import { Rodape } from '../../components/Rodape/Rodape'

export const Configuracao = () => {
  return (
    <Card>
      <section>
        <h2>Vamos come√ßar!</h2>
        <Form />
        <ListaParticipantes />
        <Rodape />
      </section>
    </Card>
  )
}
