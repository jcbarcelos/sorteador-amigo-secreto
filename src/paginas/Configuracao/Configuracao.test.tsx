/* eslint-disable testing-library/no-render-in-setup */
/* eslint-disable-next-line testing-library/no-render-in-setup */

import { render } from '@testing-library/react'
import { RecoilRoot } from 'recoil'
import { Configuracao } from './Configuracao'

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavegacao,
  }
})


describe('a pagina dfe configuração', () => {
  beforeEach(() => {
    render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>,
    )
  })
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>,
    )
    expect(container).toMatchSnapshot()
  })
})
