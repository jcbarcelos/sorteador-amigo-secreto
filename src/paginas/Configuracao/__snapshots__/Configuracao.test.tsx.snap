// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina dfe configuração deve ser renderizada corretamente 1`] = `
<div>
  <form>
    <input
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      disabled=""
      title="adicionar"
    >
      Adicionar
    </button>
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Inciar brincadeira
    </button>
  </footer>
</div>
`;
